packages/teams-js/src/public/app.ts

```diff
@@ -701,6 +701,13 @@ export namespace app {
      * Will be `undefined` when not running in Teams.
      */
     team?: TeamInfo;
+
+    /**
+     * When `processActionCommand` activates a dialog, this dialog should automatically fill in some fields with information. This information comes from M365 and is given to `processActionCommand` as `extractedParameters`.
+     * App developers need to use these `extractedParameters` in their dialog.
+     * They help pre-fill the dialog with necessary information (`dialogParameters`) along with other details.
```

Comment:
How should developers treat the case where `dialogParameters` is `undefined` differently from where it is defined but it's an empty object?

Reply:
That's a really good question, and here's the confirmation I received: We don't do anything differently when there's an empty object for dialog parameters here. Even if it's an empty array, it indicates that no key/value was passed by M365.

Reply:
Does that mean that in order for developers to know that there are no key/value parameters, they have to check BOTH that this is defined AND that the object is not empty?

Could we make their job easier by just _always_ having a defined object here and making the object empty in the case that there's no key/value pairs passed?

Reply:
`Could we make their job easier by just always having a defined object here and making the object empty in the case that there's no key/value pairs passed` that totally makes sense. 
How about this: 
we will have `dialogParameters?: Record<string, string>;` in the legacy context which send over by hubSDK, and
`dialogParameters: Record<string, string>;` in the app.context. So that we can always do `dialogParameters: legacyContext.dialogParameters || {},` in the`transformLegacyContextToAppContext` function to reduce the checking logic from app.context?

Reply:
I think there are downsides of defining the dialogParameters all the time - 
1. App developers will think they can make use of these parameters when they cannot. The parameters will be available only when M365 open a dialog for ME
2. dialogParameters should only be present in the 'task' frameContext and no where else. I am adding some logic in hubSDk to ensure this. 

I think, I would rather have developers check for undefined and empty, but also open for other opinions. Bring you votes everybody! @erinha @MengyiGong @AE-MS

Reply:
Can you say more about your concern for #1? If the object is empty, won't that be the signal to app developers that the object can't be used? If that's not what a defined empty object means, then what does it mean?

Reply:
> we will have dialogParameters?: Record<string, string>; in the legacy context which send over by hubSDK, and
> dialogParameters: Record<string, string>; in the app.context. So that we can always do dialogParameters: legacyContext.dialogParameters || {}, in thetransformLegacyContextToAppContext function to reduce the checking logic from app.context?

I like the idea Maggie, let's go with that. I also agree with Aaron's point that external app developers won't do anything differently when the object is empty vs undefined. It does not give them any extra info.

Reply:
> `Could we make their job easier by just always having a defined object here and making the object empty in the case that there's no key/value pairs passed` that totally makes sense. How about this: we will have `dialogParameters?: Record<string, string>;` in the legacy context which send over by hubSDK, and `dialogParameters: Record<string, string>;` in the app.context. So that we can always do `dialogParameters: legacyContext.dialogParameters || {},` in the`transformLegacyContextToAppContext` function to reduce the checking logic from app.context?
> 
> Thoughts? @lakhveerk

Thanks for both of your great suggestions! @AE-MS @lakhveerk I updated the `dialogParameters` to defined value in app.Context and undefined `dialogParameters` in `legacyContext`. The undefine dialogParameters from hubSDK mapped as `dialogParameters: legacyContext.dialogParameters || {}`, in thetransformLegacyContextToAppContext function. Let me know if this sounds good. :)